### PR TITLE
add support for AOI polygons

### LINKED ISSUE #270: Add support for AOI polygons
Currently, it is necessary to exhaustively label scenes for object detection. It would be nice to just label parts of the scene (AOIs) and specify those AOIs as polygons in the label file. We could implement this by using a special `class_name` for the AOI polyons, and then specifying that class name as an option for `make_training_chips` using an optional `aoi_class_name` field.

### PR DESCRIPTION
**Pull request by simonkassel**
## Overview

This PR adds support for adding an AOI polygon when creating training chips. It allows the user to use a geojson to specify the region to use when creating training and validation chips. With this added functionality, a user will be able to train a model on an image that is not exhaustively labeled.

### Checklist

- [x] Ran scripts/format_code and commited any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

## Testing Instructions

* add an `aoi_uri` to each scene in the workflow config. 
e.g.:
```
{
    "train_scenes": [
        {
            "aoi_uri":"{rv_root}/processed-data/cowc-potsdam-test/potsdam_aoi.geojson",
            "id": "2-10",
            "raster_source": {
                "geotiff_files": {
                    "uris": [
                        "{rv_root}/processed-data/cowc-potsdam-test/2-10.tif"
                    ]
                }
            },
```
* Create training chips
* run the same workflow without the `aoi_uri` 

Closes #270

### CODE REVIEW

**Comment by simonkassel** [comment]
@lewfish this is ready for you to take a look at again whenever you get a chance

File: src/rastervision/label_stores/utils.py [hunk 1]
```diff
@@ -87,3 +89,25 @@ def load_label_store_json(uri, readable):
...
+    aoi_shapely = []
+    for feature in aoi_geojson:
+        coordinates = feature['geometry']['coordinates'][0]
+        pixel_coordinates = []
+        for c in coordinates:
+            pixel_coordinate = list(crs_transformer.map_to_pixel((c[0], c[1])))
+            pixel_coordinates.append(pixel_coordinate)
+        feature['geometry']['coordinates'][0] = pixel_coordinates
+        aoi_shapely.append(shape(feature["geometry"]))
+    return aoi_shapely
```
[the remark this answers is not in the document]
Comment (simonkassel) [reply src/rastervision/label_stores/utils.py]: @lewfish you are right and I re-wrote it but for some reason that I can't figure out, i'm having a hard time when I import `get_str_tree` into utils. Any sense of why? should `get_str_tree` be in utils?

Let me know if this is not worth the time and I will just not make this change

```
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/src/rastervision/workflows/chain.py", line 27, in <module>
    from rastervision import run
  File "/opt/src/rastervision/run.py", line 3, in <module>
    from rastervision.builders import (compute_raster_stats_builder,
  File "/opt/src/rastervision/builders/make_training_chips_builder.py", line 1, in <module>
    from rastervision.builders import (scene_builder, ml_task_builder)
  File "/opt/src/rastervision/builders/scene_builder.py", line 2, in <module>
    from rastervision.builders import label_store_builder, raster_source_builder
  File "/opt/src/rastervision/builders/label_store_builder.py", line 1, in <module>
    from rastervision.label_stores.object_detection_geojson_file import (
  File "/opt/src/rastervision/label_stores/object_detection_geojson_file.py", line 7, in <module>
    from rastervision.label_stores.utils import (add_classes_to_geojson,
  File "/opt/src/rastervision/label_stores/utils.py", line 7, in <module>
    from rastervision.label_stores.classification_geojson_file import get_str_tree
  File "/opt/src/rastervision/label_stores/classification_geojson_file.py", line 11, in <module>
    from rastervision.label_stores.object_detection_geojson_file import (
ImportError: cannot import name 'geojson_to_labels'
```

[the remark this answers is not in the document]
Comment (simonkassel) [reply src/rastervision/label_stores/utils.py]: you can probably disregard this unless there's something quick and obvious that I'm missing. I addressed the other three comments in the last two commits

[the remark this answers is not in the document]
Comment (simonkassel) [reply src/rastervision/label_stores/utils.py]: ok, i made these changes, whenever you get a chance to look at them

**Comment by lossyrob** [comment]
https://travis-ci.org/azavea/raster-vision/builds/417423999#L2114

**Pull request merged**